app/queries/internal/moderators_query.rb

```diff
@@ -0,0 +1,36 @@
+module Internal
+  class ModeratorsQuery
+    DEFAULT_OPTIONS = {
+      state: :trusted
+    }.with_indifferent_access.freeze
+
+    def self.call(relation = User.all, options = {})
+      options = DEFAULT_OPTIONS.merge(options)
+      state, search = options.values_at(:state, :search)
+
+      relation = if state == "potential"
+                   relation.where(
+                     "id not in (select user_id from users_roles where role_id = ?)",
+                     role_id_for(:trusted),
+                   ).order("users.comments_count DESC")
+                 else
+                   relation.joins(:users_roles).where(users_roles: { role_id: role_id_for(state) })
+                 end
+
+      relation = search_relation(relation, search) if search.presence
+
+      relation
+    end
+
+    def self.role_id_for(role)
+      Role.find_by(name: role)&.id
```

Comment:
this can be an issue in the query on line 12. If the role is not found you should probably raise ActiveRecord::NotFound using `find_by!`. I'm saying this because the query would become `WHERE id NOT IN ())` which would return all users. 

Is that okay?

Reply:
Yeah. It makes more sense to raise an error so the user can see what's going on with the class, than making him believe that the query is returning nothing.

Thanks.